tests/Tests.elm

```diff
@@ -114,6 +140,107 @@ all =
         ]
 
 
+valid : List String
+valid =
+    [ "1.9.0"
+    , "1.10.0"
+    , "1.11.0"
+    , "1.0.0-alpha"
+    , "1.0.0-alpha.1"
+    , "1.0.0-0.3.7"
+    , "1.0.0-x.7.z.92"
+    , "1.0.0-alpha+001"
+    , "1.0.0+20130313144700"
+    , "1.352.440-111a+exp.sha.5114185"
+    , "1.352.440-beta+exp.sha.5114f85"
+    , "0.0.4"
+    , "1.2.3"
+    , "10.20.30"
+    , "1.1.2-prerelease+meta"
+    , "1.1.2+meta"
+    , "1.1.2+meta-valid"
+    , "1.0.0-alpha"
+    , "1.0.0-beta"
+    , "1.0.0-alpha.beta"
+    , "1.0.0-alpha.beta.1"
+    , "1.0.0-alpha.1"
+    , "1.0.0-alpha0.valid"
+    , "1.0.0-alpha.0valid"
+    , "1.0.0-alpha-a.b-c-somethinglong+build.1-aef.1-its-okay"
+    , "1.0.0-rc.1+build.1"
+    , "2.0.0-rc.1+build.123"
+    , "1.2.3-beta"
+    , "10.2.3-DEV-SNAPSHOT"
+    , "1.2.3-SNAPSHOT-123"
+    , "1.0.0"
+    , "2.0.0"
+    , "1.1.7"
+    , "2.0.0+build.1848"
+    , "2.0.1-alpha.1227"
+    , "1.0.0-alpha+beta"
+    , "1.2.3----RC-SNAPSHOT.12.9.1--.12+788"
+    , "1.2.3----R-S.12.9.1--.12+meta"
+    , "1.2.3----RC-SNAPSHOT.12.9.1--.12"
+    , "1.0.0+0.build.1-rc.10000aaa-kk-0.1"
+    , "1.0.0-0A.is.legal"
```

Comment:
`0.0.0--+-` would be valid, maybe interesting to add in example. But it’s not a really practical version number. :smile:

Reply:
Added because there wasn't one pinning down legality of dash in build metadata.